crates/miden-lib/asm/agglayer/account_components/bridge_out.masm

```diff
@@ -0,0 +1,86 @@
+use.miden::note
+use.miden::output_note
+
+const.BURN_NOTE_ROOT=[6407337173854817345,5626358912819151014,703918618794810515,17401169215223723177]
+
+const.EXECUTION_HINT_ALWAYS=1
+const.PUBLIC_NOTE=1
+const.AUX=0
+const.DEFAULT_FAUCET_TAG=0
+
+#! Inputs: [ASSET]
+#! Outputs: []
+proc create_burn_note.4
+    loc_storew_be.0 dropw
+
+    push.BURN_NOTE_ROOT
+    # => [SCRIPT_ROOT]
+
+    # serial num placeholder
+    push.1.1.1.1
+    # => [SERIAL_NUM, SCRIPT_ROOT]
+
+    push.0 push.0
+    # => [inputs_ptr, num_inputs, SERIAL_NUM, SCRIPT_ROOT]
+
+    exec.note::build_recipient
+    # => [RECIPIENT]
+
+    push.EXECUTION_HINT_ALWAYS
+    push.PUBLIC_NOTE
+    push.AUX
+    push.DEFAULT_FAUCET_TAG
+    # => [tag, aux, note_type, execution_hint, RECIPIENT]
+
+    call.output_note::create
+    # => [note_idx, 0,0,0]
+
+    padw loc_loadw_be.0
+    # => [ASSET, note_idx, 0,0,0]
+
+    exec.output_note::add_asset
+    # => [REQUESTED_ASSET, note_idx, 0,0,0]
+
+    dropw dropw
+    # => []
+end
+
+#! Stubbed out
+#! Inputs: [ASSET, INPUTS[1], INPUTS[0]]
+#! Outputs: [LEAF[1], LEAF[0]]
+proc mmr_frontier_keccak_add
```

Comment:
let's use the interface as outlined in https://github.com/0xMiden/miden-base/issues/2105 (which is basically the interface for `std::collections::mmr::add` (https://docs.miden.xyz/miden-vm/user_docs/stdlib/collections)